### ISSUE TITLE
DNS request blocking issue

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#2851]
There's a weird issue with the DNS filtering.

## Steps to reproduce

1. Enable DNS filtering + DNS requests blocking
2. Open `ad.doubleclick.net` in Chrome
3. Check the filtering log

## Expected result

1. `ad.doubleclick.net` is blocked on the DNS level
2. No connection from Chrome

## Actual result

1. `ad.doubleclick.net` is blocked on the DNS level
2. But there is still a connection from Chrome to that domain (to its real IP address!)

**Comment by ameshkov** [comment]
What's interesting -- it works properly when NXDOMAIN blocking is enabled (`pref.dns.blocking.nxdomain`).

We should figure what's wrong with "null IP" case.

**Comment by ameshkov** [comment]
It appears that Chrome starts using its "compression" service when DNS returns `0.0.0.0` or any loopback address.

**Comment by ameshkov** [comment]
My bad. Chrome uses it for all HTTP requests (without resolving them), but HTTPS domains are blocked properly.

**Comment by ameshkov** [comment]
Merged into #2847
